cgo/walletloader.go

```diff
@@ -217,6 +223,8 @@ func closeWallet(cName *C.char) *C.char {
 	if !exists {
 		return errCResponse("wallet with name %q does not exist", name)
 	}
+	close(w.shutdown)
```

Comment:
I'm still looking at this. This signals that wallet shutdown has been requested, but I wonder, why not just cancel the `ctx` used by most background activities including rescan? The only code that acts when this channel is closed is the rescan goroutine, and that also listens for the ctx cancellation and returns as well. An advantage to using just `ctx` cancellation would be that the rescan process itself would receive a trigger to abort.

Since `ctx` is currently used by multiple wallets, we can make each wallet have a `ctx` that is created from the general `ctx`.

Reply:
shutdown chat -> context.Context?